### PR TITLE
Refactor some code for Fennec, Fenix and iOS

### LINKED ISSUE #9819: Refactor some code for Fennec, Fenix and iOS
This supports #9714 

While working on #9714 I ended up doing quite a bit of refactoring, and the patch became quite large, so I am splitting out some of the refactoring work into this issue. The changes involve code about Fennec, Fenix and iOS.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #9819 

This is some refactoring that was going to be part of the patch for #9714. One issue is that if this patch were to land, without the patch for #9714, we would end up with some unexpected messages on Fenix. The intention is for this patch to be immediately followed by one for #9714, which will resolve any of those issues, and they would both be included in the same push.

If this is considered bad form, then I can just add these changes into the patch for #9714, but that makes it a bit larger.

### CODE REVIEW

**Comment by willdurand** [comment]
> If this is considered bad form, then I can just add these changes into the patch for #9714, but that makes it a bit larger.

I think one problem with this approach is if we need to revert the changes. Creating different commits within the same PR would help there.

File: src/amo/components/WrongPlatformWarning/index.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ export class WrongPlatformWarningBase extends React.Component<InternalProps> {
   static defaultProps = {
     _correctedLocationForPlatform: correctedLocationForPlatform,
     _getClientCompatibility: getClientCompatibility,
-    _isFenix: isFenix,
+    _isFenix: isFirefoxForAndroid,
```
Comment (willdurand) [inline src/amo/components/WrongPlatformWarning/index.js:57]: This is weird.

Comment (willdurand) [reply src/amo/components/WrongPlatformWarning/index.js:57]: Could you update the injected prop name?

**Review by willdurand — APPROVED** [review]
r+wc

**Comment by willdurand** [comment]
> I think one problem with this approach is if we need to revert the changes. Creating different commits within the same PR would help there.

I didn't try https://github.com/mozilla/addons-frontend/pull/9823 locally but I read it and the diff isn't that big so maybe you should consider merging https://github.com/mozilla/addons-frontend/pull/9823 instead of this one + https://github.com/mozilla/addons-frontend/pull/9823. As you can see, here it looks good.

File: src/amo/components/WrongPlatformWarning/index.js (hunk 1 above)
Comment (bobsilverberg) [reply src/amo/components/WrongPlatformWarning/index.js:57]: Yeah, I'm not sure what happened here. Probably an artifact of how I tried to extract some changes from a larger patch. I'll make sure this is fixed in the big patch.

**Comment by bobsilverberg** [comment]
> > I think one problem with this approach is if we need to revert the changes. Creating different commits within the same PR would help there.
> 
> I didn't try #9823 locally but I read it and the diff isn't that big so maybe you should consider merging #9823 instead of this one + #9823. As you can see, here it looks good.

Yeah, I think I'll do that and flag you for review on the larger patch.

**Pull request closed without merging**